FLIR/conservator/wrappers/media.py

```diff
@@ -204,25 +209,31 @@ def upload(conservator, upload_request):
         """
         assert isinstance(upload_request, MediaUploadRequest)
         file_path = upload_request.file_path
+        original_name = os.path.split(file_path)[-1]
         remote_name = upload_request.remote_name
         collection_id = upload_request.collection_id or None
 
         file_path = os.path.expanduser(file_path)
         assert os.path.isfile(file_path)
         if remote_name is None:
-            remote_name = os.path.split(file_path)[-1]
+            remote_name = original_name
 
         media = None
         try:
-            media = MediaType._create(conservator, remote_name, collection_id)
-            upload_id = media._initiate_upload(remote_name)
+            media = MediaType._create(conservator, original_name, collection_id)
+            if remote_name and (original_name != remote_name):
```

Comment:
Does the "name" field match "filename" when the media file is created?  Or should I remove the compare between `original_name` and `remote_name`?

Reply:
No. `name` is not set when a video is created. Remove lines 218/219.